### PR TITLE
Add dd() as quick alternative for debug(); die(); 

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves https://github.com/cakephp/cakephp/issues/9938

I checked, laravel etc use also dd() function, so coming from them it will also feel a bit easier to work the way one might have used to work :)

### CODE REVIEW

**Comment by markstory** [comment]
Looks good to me.

**Comment by thinkingmedia** [comment]
@dereuromark Can this have the same return variable behavior has `debug()`.

For example;

```
$document = dd($this->Document->get($id));
```

**Comment by dereuromark** [comment]
Why? The point is to die() here.

**Comment by thinkingmedia** [comment]
Can this have a conditional argument the defaults to `true` if `dd()` should die?

For example;

```
$document = dd($this->Document->get($id), $foo === 'something');
```

It would print `debug()` output always, but die if the condition is true.

I know this would interfere with the showHtml parameter, but I'll leave that up to you :)

**Comment by thinkingmedia** [comment]
```
$document = dd($this->Document->get($id));
```

Would die only if debug is enabled. ;)

**Comment by dereuromark** [comment]
And otherwise be completely ignored (for accidental keepings on the live server to not kill anything).

**Comment by thinkingmedia** [comment]
That's how `debug()` works now. You can do this and push to production without worry.

```
$document = debug($this->Document->get($id));
```

**Comment by dereuromark** [comment]
dd() is void by design. You shouldnt try to check on any return value here.

**Comment by markstory** [comment]
Would it be reasonable if dd() always died?

**Comment by ADmad** [comment]
IMO dd() should be identical to debug() with just an additional die() at the end.

**Comment by dereuromark** [comment]
It might be weird to be not consistent with the other debug functions.

**Comment by ADmad** [comment]
> It might be weird to be not consistent with the other debug functions.

In only in consistency with my proposal will be lack of return value since there will be a die() at end of function.

```php
function debug($var, $showHtml = null, $showFrom = true)
    {
        if (!Configure::read('debug')) {
            return;
        }
        debug($var, $showHtml, $showForm);
        die();
    }
```

**Comment by dereuromark** [comment]
Why are we continuing discussing semantics now, when the solution is already there with a single merge click? Bottom line: It works as expected, problem solved. We should now all focus again on the real issues at hand IMO :)

**Comment by ADmad** [comment]
Why did you make this PR instead of focusing on "real issues at hand"?

Closing the PR is also a single click away. I can make a new one which is more suitable. At least 2 other team members have upvoted my earlier comment. I don't see any in favor of merging this PR in it's current form.

**Comment by dereuromark** [comment]
I disagree, most are in sync of this PR. I think you produced issues for no reason just now. Why?

Again for you:

a) I shouldnt return because that invites for keeping it longer than necessary, It is and always should be a very tmp thing.

b) just debug() and die() is nonsense, as it cannot print the source location out, thus the necessary changes on top here.

> Then why did you make this PR instead of focusing on "real issues at hand"?

Please be a lit less offending in your way of producing claims. This is PR was fine until you now started to get involved. Or make your 2nd PR and let's see which one is the one the whole community would like to go with.

**Comment by dereuromark** [comment]
It *could* have been:

- PR
- OK, does the job (markstory: "Looks good to me")
- Merge
- No discussion necessary

....

**Comment by ADmad** [comment]
> just debug() and die() is nonsense, as it cannot print the source location out, thus the necessary changes on top here.

So this should have been your response instead of trying to short circuit the discussion.

File: src/basics.php [hunk 1]
```diff
@@ -151,3 +151,28 @@ function breakpoint()
...
+     * @link http://book.cakephp.org/3.0/en/development/debugging.html#basic-debugging
+     */
+    function dd($var, $showHtml = null)
+    {
+        if (!Configure::read('debug')) {
+            return;
+        }
+
+        debug($var, $showHtml, false);
+        $backtrace = debug_backtrace(false, 1);
```
Comment (inoas) [inline src/basics.php]: Couldn't these 2 lines added to function `debug()` instead?
Would that make @ADmad happy?

Comment (dereuromark) [inline src/basics.php]: Feel free to further fine tune with future PRs.

**Pull request merged**